zerver/data_import/microsoft_teams.py

```diff
@@ -25,9 +26,169 @@
+    visibility: Literal["public", "private"]
+    is_archived: bool
+    zulip_channel_id: int
+    zulip_recipient_id: int
+
+
+AddedTeamsT: TypeAlias = dict[str, TeamMetadata]
+TeamIdToZulipRecipientIdT: TypeAlias = dict[str, int]
 MicrosoftTeamsUserIdToZulipUserIdT: TypeAlias = dict[str, int]
 MicrosoftTeamsFieldsT: TypeAlias = dict[str, Any]
 
+MICROSOFT_TEAMS_DEFAULT_ANNOUNCEMENTS_CHANNEL_NAME = "All company"
+
+
+def add_custom_announcements_channel(
+    realm: dict[str, Any],
+    realm_id: int,
+    default_announcement_channel_id: int | None,
+    microsoft_teams_user_id_to_zulip_user_id: MicrosoftTeamsUserIdToZulipUserIdT,
+) -> None:
+    channel_id = default_announcement_channel_id
+
+    # A team called "All Company" is always included in the export data
+    # regardless of whether the organization actually has or uses it.
+    # In the chance where it is used, we'll use that as the default
+    # announcement channels. If not, create an artificial one.
+    if default_announcement_channel_id is None:
+        recipient_id = NEXT_ID("recipient")
+        channel_id = NEXT_ID("channel")
+        channel = build_stream(
+            # Microsoft Teams export doesn't include teams creation date.
+            float(timezone_now().timestamp()),
+            realm_id,
+            MICROSOFT_TEAMS_DEFAULT_ANNOUNCEMENTS_CHANNEL_NAME,
+            "This is the default group for everyone in the network",
+            channel_id,
+            False,
+            False,
+        )
+        realm["zerver_stream"].append(channel)
+        channel_id = channel["id"]
+
+        recipient = build_recipient(channel_id, recipient_id, Recipient.STREAM)
+        realm["zerver_recipient"].append(recipient)
+
+        all_user_ids = list(microsoft_teams_user_id_to_zulip_user_id.values())
+        for user_id in all_user_ids:
+            sub = build_subscription(
+                recipient_id=recipient_id,
+                user_id=user_id,
+                subscription_id=NEXT_ID("subscription"),
+            )
+            realm["zerver_subscription"].append(sub)
+
+    realm["zerver_realm"][0]["new_stream_announcements_stream"] = channel_id
+    realm["zerver_realm"][0]["zulip_update_announcements_stream"] = channel_id
+    realm["zerver_realm"][0]["signup_announcements_stream"] = channel_id
+    logging.info("Using the channel 'All Company' as default announcements channel.")
+
+
+def convert_teams_to_channels(
+    microsoft_teams_user_id_to_zulip_user_id: MicrosoftTeamsUserIdToZulipUserIdT,
+    realm: dict[str, Any],
+    realm_id: int,
+    teams_data_dir: str,
+) -> AddedTeamsT:
+    team_data_folders = []
+    for f in os.listdir(teams_data_dir):
+        path = os.path.join(teams_data_dir, f)
+        if os.path.isdir(path):
+            team_data_folders.append(f)
+
+    logging.info("######### IMPORTING TEAMS STARTED #########\n")
+
+    # Build teams subscription map
+    raw_teams_subscription_data: dict[str, list[int]] = defaultdict(list)
```

Comment:
I think we can name this variable less vaguely, e.g. `team_id_to_zulip_subscriber_ids` or such.